build/aten/src/ATen/Functions.h

```diff
@@ -2238,12 +2239,15 @@ static inline Tensor & baddbmm_out(Tensor & out, const Tensor & self, const Tens
 #endif
 }
 static inline Tensor bartlett_window(int64_t window_length, const TensorOptions & options) {
+    return bartlett_window(window_length, typeMetaToScalarType(options.dtype()), options.layout(), options.device(), options.pinned_memory());
```

Comment:
Are we still going to codegen this line? Or will it be added to the Functions.h template?

If we end up codegen-ing this line, we'll still have a special case for TensorOptions in the code generation (but it'll probably look nicer than before). 

If we stop codegen-ing this line, then we'll have to write it manually in the Functions.h template (which is not a bad thing imo because these rarely change) and we can get rid of all TensorOptions handling in the codegen

Reply:
I'll see whats better but im gravitating more towards generation as its more flexible and once we add a proper support for 'requires_grad', i would like to change less places rather than more. but will see